lib/src/bloc/dice_bloc.dart

```diff
@@ -0,0 +1,40 @@
+import 'dart:async';
+import 'dart:math';
+import '../styles/constants.dart';
+import 'i_dice_bloc.dart';
+
+class DiceBloc extends IDiceBloc {
+  List<int> _diceValues = [];
+  StreamController<List<int>> _streamValueController =
+      StreamController<List<int>>.broadcast();
+  StreamController<String> _streamDicesScoreController =
+      StreamController<String>.broadcast();
+
+  @override
+  void dispose() {
+    _streamValueController.close();
+    _streamDicesScoreController.close();
+  }
+
+  @override
+  void initialize() {}
+
+  @override
+  void roll() {
+    this._diceValues.clear();
+    int score = 0;
+    for (int i = 0; i < Constants.diceCount; i++) {
+      int randomValue = Random().nextInt(Constants.diceFaces) + 1;
```

Comment:
randomValue is the same like score, add '_' at beginning, and try to declare out of the loop. If you declare inside for you're creating a new variable for every loop

Reply:
done. Thank you